File: src/main/java/intro/to/design/patterns/using/javalang/templatemethod/StringDisplay.java
package intro.to.design.patterns.using.javalang.templatemethod;

public class StringDisplay extends AbstractDisplay {
    private final String string;
    private final int width;

    public StringDisplay(String string) {
        this.string = string;
        this.width = string.length();
    }

    private void printLine() {
        System.out.print("+");
        for (int i = 0; i < width + 2; ++i) {
            System.out.print("-");
        }
        System.out.println("+");
    }

    @Override
    public void open() {
        printLine();
    }

    @Override
    public void print() {
        System.out.println("| " + string + " |");
    }

    @Override
    public void close() {
        printLine();
    }
}

File: typescript-design-pattern/src/into-to-design-patterns-using-java-lang/template-method/string-display.ts
import Display from './display'
import ResultString from './result-string'

export default class StringDisplay extends Display {
  private readonly value: string
  private readonly width: number
  private readonly fileStore: ResultString

  constructor(value: string, fileStore: ResultString) {
    super()
    this.value = value
    this.width = value.length
    this.fileStore = fileStore
    this.fileStore.start('StringDisplay')
  }

  private printLine(): void {
    this.fileStore.write('+')
    for (let i = 0; i < this.width + 2; ++i) {
      this.fileStore.write('-')
    }
    this.fileStore.write('+')
  }

  open(): void {
    this.printLine()
    this.fileStore.write('\n')
  }

  print(): void {
    this.fileStore.write(`| ${this.value} |\n`)
  }

  close(): void {
    this.printLine()
  }
}
